src/Microsoft.OData.Core/UriParser/Parsers/TypeParsers/Common/IUriTypeParser.cs

```diff
@@ -0,0 +1,43 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="IUriTypeParser.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+namespace Microsoft.OData.Core.UriParser.Parsers.TypeParsers.Common
+{
+    #region Namespaces
+
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Linq;
+    using System.Text;
+    using Microsoft.OData.Edm;
+
+    #endregion
+
+    /// <summary>
+    /// Interface for UriTypeParser.
+    /// To parse the uri of an OData request into objects, the ODataUriParser uses UriTypeParses.
+    /// Implementation of this interface will parse a text of an EdmType to it's instance.
+    /// </summary>
+    public interface IUriTypeParser
+    {
+        /// <summary>
+        /// Parse the given text of EdmType 'targetType' to it's object instance.

```

Comment:
use <paramref name="targetType"/>.
same for parsingException.

same for other files